level05/CoffeeMachineTests.hs

```diff
@@ -275,28 +273,16 @@ cRefundCoins mach = Command gen exec
     exec :: RefundCoins Concrete -> m Int
     exec _ = evalIO (C.refund mach)
 
+-- TODO
+--
+-- In this command you'll want to set the drink cost and the
+-- credit-checked value, with appopriate callbacks so this command can
```

Comment:
`appopriate` typo.

Also is it important that it only happens once? I think we've configured the tests so that a dispense can only happen once, but do we allow for the model to change enough over time that was once TooLittle credit is now Enough?

Reply:
Good question. I think the correct thing might be to have adjusting the drink erase the drink cost/credit check result (or use a require so it can happen only once).

The alternative is structuring the tests so you have two phases: a configuration phase, then a check-and dispense phase.

Reply:
I think updating the model to erase the check result is the better option, that would test things more thoroughly, I think.

Having the separate phases for testing is a good thing but it might be overkill for this? I worry about something like that ending up as a refactoring exercise, for us or participants..

Reply:
Depends on whether or not we say "this is a bit artificial, but we want to demonstrate the principle" or we want to test this system in the most appropriate way.

Reply:
Yeah, nah that works.. We could set up the level to be:

- `CoffeeMachineTests`
- `CoffeeMachineTests.DrinkConfig`
- `CoffeeMachineTests.DrinkDispense`

Or however that yak ends up being shaved... and then call it out as you say. Thoughts?

Reply:
Nah yeah m8